mods/fireflies/init.lua

```diff
@@ -27,6 +27,58 @@ minetest.register_node("fireflies:firefly", {
 	floodable = true,
 	on_flood = function(pos, oldnode, newnode)
 		minetest.add_item(pos, "fireflies:firefly 1")
+	end,
+	on_place = function(itemstack, placer, pointed_thing)
+		local player_name = placer:get_player_name()
+		local pos = pointed_thing.above
+
+		if not minetest.is_protected(pos, player_name) and
+				not minetest.is_protected(pointed_thing.under, player_name) and
+				minetest.get_node(pos).name == "air" then
+			minetest.set_node(pos, {name = "fireflies:firefly"})
+			minetest.get_node_timer(pos):start(1)
+			itemstack:take_item()
+		end
+		return itemstack
+	end,
+	on_timer = function(pos, elapsed)
+		if minetest.get_node_light(pos) > 11 then
+			minetest.set_node(pos, {name = "fireflies:hidden_firefly"})
+		end
+		minetest.get_node_timer(pos):start(30)
+	end
+})
+
+minetest.register_node("fireflies:hidden_firefly", {
+	description = "Hidden Firefly",
+	drawtype = "airlike",
+	inventory_image = "fireflies_firefly.png",
+	wield_image =  "fireflies_firefly.png",
+	paramtype = "light",
+	sunlight_propagates = true,
+	walkable = false,
+	pointable = false,
+	diggable = false,
```

Comment:
I think it should be buildable_to to avoid an invisible and unremovable node that stops building at a location (which will seem like a bug).

Reply:
Not sure 'diggable = false' is needed since it is not in any dig group.

Reply:
Added buildable_to = true

Reply:
Also unsure if "diggable = false," is needed, but default air includes it. Default air also includes "floodable = true" and "air_equivalent = true". I can't even find "air_equivalent" in lua_api to check that one and I can see from testing this node floods as expected.

Reply:
'diggable = false' and 'drop = ""' are probably not needed since not pointable or in any dig group.

Reply:
Oops sorry nevermind the repeated comment.
'diggable = false' is harmless and might actually avoid digging through use of 'dig_node()', so no problem.